### DISCUSSION TITLE
G7 does not pair with Samsung A-Series

### DISCUSSION THREAD

**Discussion by oracess** [NightscoutFoundation/xDrip#2902] [Q&A] [answered]
G7 does not pair with Samsung A-Series! 

![Not-Autho](https://github.com/NightscoutFoundation/xDrip/assets/41123950/94a2b4ee-387c-450b-93c6-7784ac4ee91f)

Is here a fight between Dexcom and Xdrip+ ?? Dexcom App and BYODA App can pair Bluetooth. My only solution is to pair with Dexcom or Byoda app, after pairing and deleting Dexcom or Byoda app including cache and data - start handy new and then xdrip+ will work within 1 minute.

![Screenshot_20230622_135450_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/41123950/57c2b60d-d298-4db1-9dda-41cbbf5c8255)

Please help me to make a solution for this problem, I can`t believe that is only a problem for me. My handy is a Samsung A53 with Android 13 and Xdrip+ nightly d46b342-2023.06.17 Thanks to all

Edit:
Be careful with "01 June 2023 Android 13 Update", new security features

**Comment by oracess** [comment, marked as answer]
@jamorham

Another error in G5/G6/G7 status.
Since 2019, inactive Bluetooth is hidden in the status bar on Android 12/13 when BT is switched on!
But this also means that the BT request is not recognized by either Xdrip+ or Android 12/13, but is still active!
The obligatory message in the "G5/G6/G7 status" then appears: "Turn on BT and try again"
Furthermore, incorrect values ​​such as data source, Bluetooth device, connection status and sensor start (40d0h) in the case of external coupling are also displayed in the "Classic Status Page".
It is therefore possible that this can lead to coupling problems and only external coupling as described in #2902 can lead to coupling.

![status](https://github.com/NightscoutFoundation/xDrip/assets/41123950/874a47f1-bfb5-4521-ab67-9d31b2b3769b)

**Comment by jamorham** [comment]
Please try the latest nightly to see if it resolves the problem.

**Reply by oracess** [reply]
Thanks for answering. Last test was 4 days ago with the nightly from 2nd July and it only paired with workaround, so I will test it again with new sensor next week. Sorry for my posts but I have more problems as "not pairing":

1) A53 5G does not find available BT device (sensor), A51, A52, A54 does. A53 5G is in warranty repair to see if BT is defective, but all Samsung A-Series cannot pair with Xdrip+

2) Dexcom G7 sensors are 70% faulty (no 10 days + 12 hours), just fail (replace sensor)
or are temporarily unavailable (please wait 30 minutes) or cannot be started at all because the sensor bristle is outside the needle.

3) G7 less range, miserable BT. My A53G obviously miserable BT but not completely defective.
etc. etc. and so on

**Reply by jamorham** [reply]
Thanks just let me know when it is practical to test.

1) Your problem with the A53 I suspect is to do with the June security patch which has broken its bluetooth over many different apps that we have heard reported, see: https://github.com/NightscoutFoundation/xDrip/discussions/2905 for some more discussion of this. Hopefully Samsung will release a software fix.

2 & 3) The problems you report for these points have been seen by many others and especially the poor range has been widely noted. Hopefully the reliability and accuracy will be improved at the factory over time. Of course none of that helps you today.

**Reply by oracess** [reply]
Have now changed the sensor with the Nightly from 2023.07.13, pairing does not occur after 60 minutes,
then paired with Dexcom app and taken over with Xdrip. See Figure 5.
Figure 1-4 shows the pairing attempt with all error messages. An A51 was used for testing, as the A53 is under repair because of BT.
Will continue to pursue it and continue to apply my patch, not so great but does everything I want.

![1_Bild](https://github.com/NightscoutFoundation/xDrip/assets/41123950/a2fe663e-1e38-45f9-8b57-dfe7a80a3ca8)
![2_Bild](https://github.com/NightscoutFoundation/xDrip/assets/41123950/6888b236-5ee6-4221-b8f5-6993c04edd76)
![3_Bild](https://github.com/NightscoutFoundation/xDrip/assets/41123950/93e34c5c-dfbc-45b3-982c-a3e605b3e55d)
![4_Bild](https://github.com/NightscoutFoundation/xDrip/assets/41123950/14157acd-871a-47db-aa5a-01d45fc86b1c)
![5_ Bild](https://github.com/NightscoutFoundation/xDrip/assets/41123950/090a72f2-a3e0-4c1b-98dd-9bec88513d4f)

**Comment by oracess** [comment]
Samsung A53 5G is back from repair. Nothing was wrong only the firmware was updated, but this will come officially mid of september for all.
In the meantime I got Dexcom G7 Rev 00012. This Revision is better in bluetooth range but nothing more.
Pairing with xdrip+ will still not work! Workaround with dexcom app is still necessary. The mistake must be in Xdrip+ because pairing is possible with BYODA and Dexcom and this within seconds. When paired you can delete BYODY or Dexcom app and Xdrip+ will work (406a925-2023.08.17)
Still waiting for solution !

**Comment by jamorham** [comment]
Can you enable OB1 debugging as described here: https://github.com/NightscoutFoundation/xDrip/wiki/Enabling-Debugging-for-OB1-collector and then submit logs next time you have the problem with pairing and let me know when you have done so. Keep an eye on the status page while it is trying to pair to see if any useful information is highlighted.

**Reply by oracess** [reply]
I have sent logfile, pairing is working, only warmup time is 30 not 120 minutes but data comes within  30 minutes.
Thanks for all oracess

**Reply by rudysroost** [reply]
I must be missing something. The only difference is my device is a Pixel 7. I guess I'll keep trying. Thank you @oracess .

**Reply by oracess** [reply]
@rudysroost 
Before you start searching be sure that xdrip is (c680a19-2023.08.23) then enable OB1 debugging as described here: 

[https://github.com/NightscoutFoundation/xDrip/wiki/Enabling-Debugging-for-OB1-collector](url)

Best and easy with QR Code, then restart handy and look at (View Events Log) and search for "Received Authentication" this must have a value. If cannot find something unusual you must start from begin

[https://navid200.github.io/xDrip/docs/Dexcom/G7.html](url)

Be sure that other unusual things still consists as Classic Status Page, Warming Up 120 insteed 30 min (but be ready at 30 min) sometimes  loss of BT-Meter or Basis Bolus Novopen together no transmission of 2nd Pen to nightscout etc. - but on the hole it will work.

Greetings from Austria

**Comment by oracess** [comment]
@rudysroost
You can follow here #2902  When I have sent the logfile to jamorham, I will tell you if we can stop all 10 days workaround.
oracess

**Comment by oracess** [comment]
@jamorham 
For the problem "Warmup" see picture, but I think the "Status Page" for G7 must be completely revised. See picture above "Systemstatus" nothing is correct without Version.

Also a problem is two Novopen 6 (Basis/Bolus) In Xdrip+ Homescreen is all OK, but transer to Nightscout (NightscoutUploader) for Basis Novopen 6 fails, for Bolus Pen is correct, but for correct transfer you can see no entry in logfile. See picture NightscoutUploader from Log.

![Screenshot_20230826_221615_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/41123950/d4fa9da5-9806-43a8-80f2-d3d55766fa36)
![Screenshot_20230826_221042_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/41123950/b2fb03bf-5f18-4376-b7c0-69115ebb8978)
![Screenshot_20230826_221435_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/41123950/03f4c459-fad7-4cc3-80de-5084e35ff28d)

**Comment by oracess** [comment]
@jamorham
New set sensor Rev0009 with .apk from 2023.08.27 failed pairing nothing changed in Xdrip+, log was sent. Before set sensor Rev0012 paired with .apk from 2023.08.23 new sensor Rev0009 not.  I returned back to use workaround pairing with dexcom app, it still works. How can I change from old sensor to new sensor and get new sensor pairing. Changing "Transmitter ID" only from old to new definitly not working.

I have sent  two logfiles:
1) No Pairing with new installed Xdrip+ from 2023.08.27, Dexcom G7 Rev 0009, A53 5G with latest BT Patch
2) Pairing with Dexcom App (Workaround) = functional

Misfunction Novopen 6 is resolved. Nightscout is not able in standard mode to receive data for long acting insulin in my case Toujeo. I have Basis/Bolus therapy. I have to discuss this with https://ns.10be.de/ my nightscout provider.

**Comment by oracess** [comment]
@jamorham

Das Selbe wie immer, kein Pairing mit Xdrip, Workaround mit Original App ist notwendig. Genutzte Nightly Version = Nightly build 8th Sep 2023 ohne Erfolg. Nochmals erwähnt sei, dass noch immer Dexcom Version 8 und 9 in Europa ausgliefert wird, obwohl die letzte Version 14 ist und wesentlich besser und stabiler als die 8,9er werkelt. Version 14 habe ich mir von New York mitgenommen, Version 12 ist die aktuelle Version für Europa und man sollte darunterliegende Versionen sofort zurück schicken, wenn nicht als Testsensor erhalten.
Ich bekomme die Sensoren zum Laufen, aber es ist außergewöhnlich gefährlich diese Konstellation zu benutzen. Ich hoffe man geht dieser Situation bald auf den Grund, siehe auch was Nutzer auf insulinclub.de berichten.

Gruss oracess

Same as always, no pairing with Xdrip, workaround with original app is necessary. Used Nightly Version = Nightly build 8th Sep 2023 without success. It should be mentioned again that Dexcom versions 8 and 9 are still being delivered in Europe, although the last version is 14 and works much better and more stable than the 8.9. I took version 14 with me from New York, version 12 is the current version for Europe and you should send back the underlying versions immediately if you don't receive them as a test sensor.
I can get the sensors to work, but it's exceptionally dangerous to use this setup. I hope we get to the bottom of this situation soon, see what users report on insulinclub.de/index.php?thread/34242-erfahrungsberichte-g7/&pageNo=33.

Greetings oracess

**Comment by oracess** [comment]
Log was sent with #2902
